userenvs/fedora31.json

```diff
@@ -0,0 +1,41 @@
+{
+    "workshop": {
+	"schema": {
+	    "version": "2020.03.02"
+	}
+    },
+    "userenv": {
+	"name": "fedora31",
+	"label": "Fedora 31",
+	"origin": {
+	    "image": "docker.io/library/fedora",
+	    "tag": "31"
+	},
+	"properties": {
+	    "packages": {
+		"type": "rpm",
+		"manager": "dnf"
+	    }
+	}
+    },
+    "requirements": [
+	{
+	    "name": "python36",
```

Comment:
I think we are going to a need a "post-install" option to run commands after a requirement is installed.  For example, this will be necessary to support python-3 for RHEL/CentOS 7, where we need to run a command to make the default python interpreter version 3 (using alternatives utility).

Reply:
Wouldn't that just be a "manual" section with the commands listed?

Reply:
For example (purely made up scenario):

```
{
    "name": "distro-fio",
    "type": "distro",
    "distro_info": {
        "packages": [
            "fio"
        ]
    }
},
{
    "name": "fio-version-check",
    "type": "manual",
    "manual_info": {
        "commands": [
            "fio --version"
        ]
    }
}
```

Reply:
I am not sure.  I guess I need to see an example where this would work for rh-python-36 installed via the distro section for rhel7 with the manual step to set python3 as "/usr/bin/python".

The other option is that we make 100% sure all of our scripts use /usr/bin/python3 and confirm all distros we use, when a version 3.x python is installed, it always installs a /usr/bin/python3

Reply:
I'm not opposed to doing the rhel7 stuff but...

Maybe just setting the scripts to use `/usr/bin/python3` is the way to go.  From a quick survey none of my Fedora 31, RHEL8, or CentOS 8 VMs have `/usr/bin/python` by default (I think, hard to be sure as there are a few that have it but I'm pretty sure that is because I manually did something).